### PR TITLE
Happy new year 2019

### PR DESCRIPTION
**Pull request by Quetzacoalt91**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | develop
| Description?  | This is that time of the year again, where we must update our license header. :champagne: 
| Type?         | refacto
| Category?     | CO
| BC breaks?    | Nope
| Deprecations? | Nope
| Fixed ticket? | /
| How to test?  | Only the file headers have changed. Features remain untouched.

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/11997)

### CODE REVIEW

**Review by PierreRambaud — DISMISSED** [review]

**Review by matks — DISMISSED** [review]

**Comment by mickaelandrieu** [comment]
Sounds like Symfony project have updated again his file header: https://github.com/symfony/symfony/blob/master/src/Symfony/Component/BrowserKit/Client.php#L5

Never too late to do the right thing 👼 

imho only the license file should change if you want to keep the date on it, but according to the [OSI](https://opensource.org/licenses/OSL-3.0), this is not a requirement? /c @ttoine 

I won't reject this pull request, it's up to you, mates 👍

File: src/PrestaShopBundle/Command/UpdateLicensesCommand.php [hunk 1]
```diff
@@ -56,7 +56,7 @@ class UpdateLicensesCommand extends Command
  * needs please refer to http://www.prestashop.com for more information.
```
Comment (MathiasReker) [inline src/PrestaShopBundle/Command/UpdateLicensesCommand.php]: Missing s in protocol -> `https://www.prestashop.com`

**Comment by MathiasReker** [comment]
```
#!/bin/sh
updateYear() {
  NEW_YEAR=`date +'%Y'`
  BRANCH_NAME="showcase-happy-new-year-$NEW_YEAR"
  if [[ ! `git branch --list "$BRANCH_NAME"` ]]; then
    git checkout "develop"
    git checkout -b "$BRANCH_NAME"
    array=('PHP' 'JS' 'TXT' 'TPL' 'CSS' 'SCSS' 'MD' 'TWIG' 'VUE')
    for FILE_TYPES in "${array[@]}"
    do
      OLD_YEAR=2014 #check years older than last year
      while [[ $OLD_YEAR < $NEW_YEAR ]]; do
        find ./ -iname "*.$FILE_TYPES" -exec sed -i "/$OLD_YEAR PrestaShop/ s/$OLD_YEAR PrestaShop/$NEW_YEAR PrestaShop/g" '{}' ';'
        OLD_YEAR=`expr $OLD_YEAR + 1`
      done
      if ! git diff-index --quiet HEAD --; then
        git add -A
        git commit -m "Update license of $FILE_TYPES files"
      fi
    done
  fi
  echo ""
  echo "Done!"
}
updateYear
```

I think you missed some file extensions and maybe years older than 2018? This shell script explain what I mean :)

**Comment by Quetzacoalt91** [comment]
The command `prestashop:license:update` checks the whole LICENSE block exists, adds it if it is missing or replaces it otherwise.

This allow us to not take care about the previous year, it should get the 2019 at all times.

> I think you missed some file extensions

With the extension, do you have a concrete example in mind by any chance?

**Comment by MathiasReker** [comment]
```
\docs\licences\LICENSE.txt
\LICENCE.md
\docs\CHANGELOG.txt
\admin-dev\themes\new-theme\scss\components\_form.scss

```

**Comment by mickaelandrieu** [comment]
Hi @Quetzacoalt91,

would you mind to rebase so I can merge this one?

Thanks mate 👍

**Comment by Quetzacoalt91** [comment]
PR rebased

**Review by PierreRambaud — DISMISSED** [review]

**Pull request merged**